x-pack/test/functional/apps/ml/anomaly_detection_jobs/custom_urls.ts

```diff
@@ -100,7 +101,7 @@ export default function ({ getService }: FtrProviderContext) {
     it('adds a custom URL to Dashboard in the edit job flyout', async () => {
       await ml.jobTable.addDashboardCustomUrl(JOB_CONFIG.job_id, testDashboardCustomUrl, {
         index: 1,
-        url: `dashboards#/view/${testDashboardId}?_g=(filters:!(),time:(from:'$earliest$',mode:absolute,to:'$latest$'))&_a=(filters:!(),query:(language:kuery,query:'airline:\"$airline$\"'))`,
+        url: `dashboards#/view/${testDashboardId}?_g=(filters:!(),time:(from:'$earliest$',mode:absolute,to:'$latest$'))&_a=(filters:!(('$state':(store:appState),meta:(alias:fq_multi_1_custom_urls,disabled:!f,index:'${dataViewId}',negate:!f,type:custom,value:'{"bool":{"must":[{"match_all":{}}]}}'),query:(bool:(must:!((match_all:())))))),query:(language:kuery,query:'airline:"$airline$"'))`,
```

Comment:
❓ if the query is `match_all` (i.e. the default), do we actually want it to end in the URL?

Reply:
Good call. Originally I thought for consistency, we should just add it anyway. But since this is a pretty common DSL default query, I added a check to handle it here https://github.com/elastic/kibana/pull/141871/commits/21b3a249907fc77d5dfca985a8d452911ef04d8b